src/zope/securitypolicy/tests/test_securitymap.py

```diff
@@ -208,3 +208,17 @@ class ASM(AnnotationSecurityMap):
         self.assertIn(ASM.key, context.annotations)
         psm = context.annotations[ASM.key]
         self.assertIs(psm, sec_map.map)
+
+
+class TestSecurityMapInheritance(unittest.TestCase):
+
+    def test_inheritance(self):
+
+        from zope.securitypolicy.securitymap import SecurityMap
+        class CustomSecurityMap(SecurityMap):
+
+            def __nonzero__(self):
```

Comment:
The argument could be made that a class that implements `__nonzero__` but *not* `__bool__` is just broken on Python 3: if you want to support Python 3, you need to implement `__bool__`. It looks funny to have a class that has `__nonzero__` but not `__bool__`, and if I encountered it, the first thing I would do would be to define `__bool__`. I shouldn't need to know that a super class's `__bool__` delegates to a non-standard, Python 2-only dunder method.

It's not clear to me that we should enshrine `__nonzero__` as the canonical way for subclasses of `SecurityMap` to implement `bool()` on Python 3 in this way. 

OTOH, I can understand an argument from the practicality standpoint, if there are a large number of such subclasses that aren't actually tested on Python 3 and so would be subtlety broken: this slightly bandaids the pain. Testing them on Python 3 would reveal the problem, though, whereupon it could be fixed.

Reply:
So this fixes subclasses that override `__nonzero__` and forget `__bool__`, but keeps subclasses that override `__bool__` and forget `__nonzero__` broken.

I suppose the argument is that subclasses that override `__nonzero__` and don't touch `__bool__` are more likely due to legacy Python-2-only code than subclasses that override `__bool__` and forget about `__nonzero__`.

I'm -0 for this change, now that I understand the motivation.

Reply:
Yeah, I made the request for backwards compatible reasons, but if the new behaviour is the desired one we should of course let it remain.
I wonder if we can come up with a way to make a deprecation warning?